### PR TITLE
Fixed GitHub authentication - #1764

### PR DESCRIPTION
**Pull request by adeeb1**
Only public email addresses are returned with the call to `/user`, which returns a value of `null` if a GitHub user has no specified public email address. Utilized the `/user/emails` call to retrieve the user's primary private email address to ensure that an email address would be available. For more info, see [this post](http://stackoverflow.com/questions/35373995/github-user-email-is-null-despite-useremail-scope/35387123#35387123)

### CODE REVIEW

File: server/routes/github.coffee [hunk 1]
```diff
@@ -28,26 +28,38 @@ module.exports.setup = (app) ->
...
-          if not emailLower
-            return errors.serverError res, "Problem finding GitHub user with that identity."
+          
+          request.get {uri: 'https://api.github.com/user/emails', headers: headers}, (err, r, response) ->
+            return log.error err if err?
+            githubUserEmails = JSON.parse response
+            log.info 'Got GitHub user emails', githubUserEmails
+          
+            for email in githubUserEmails
+              if email.primary == true
```
Comment (sderickson) [inline server/routes/github.coffee]: Use `is` instead of `==` in CoffeeScript. Alternatively, use [_.find](https://lodash.com/docs#find).

**Comment by sderickson** [comment]
Changes look good except for the minor tweak. The button is not incorporated with our design, however, so we're going to need to hide the button still for now until we figure out how to fit it into our interface. But this should make it easier to do so! :+1:

**Comment by adeeb1** [comment]
Thank you for the feedback, @sderickson! I've implemented your tweaks and hidden the GitHub login button.

**Comment by sderickson** [comment]
Merged it in but squashed it, so closing this manually. Thanks!

Hmm, this might cause problems with your fork. Please let me know if it does. Trying to figure out the best way to keep history clean but also not get people into git hellholes.

**Comment by adeeb1** [comment]
Awesome, glad to help!

I've synced my fork without any trouble, but it says that I'm four commits ahead of the original branch. It lists my commits as the ones that are ahead.

**Pull request closed without merging**